language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ErrorConstructorExpressionNodeContext.java

```diff
@@ -201,8 +219,22 @@ private boolean isInErrorMessageArgContext(BallerinaCompletionContext context,
                 && textRange.startOffset() <= cursor && cursor <= textRange.endOffset());
     }
 
+    private boolean isInErrorCauseArgContext(BallerinaCompletionContext context,
+                                             ErrorConstructorExpressionNode node) {
+
+        if (!withinArgs(context, node) || node.arguments().isEmpty() || node.arguments().size() == 1) {
+            return false;
+        }
+        FunctionArgumentNode secondArgNode = node.arguments().get(1);
+        TextRange textRange = secondArgNode.textRange();
+        int cursor = context.getCursorPositionInTree();
+        return cursor < textRange.startOffset() || (secondArgNode.kind() == SyntaxKind.POSITIONAL_ARG
+                && textRange.startOffset() <= cursor && cursor <= textRange.endOffset());
```

Comment:
Shall we remove this condition and instead check if the cursor is before the `close paran`. 

```
                cursor <= textRange.endOffset());
```

The reason is, when there is more than two params this would not work.

Eg: 
error("message", namedArg1 = val, <cursor>)

Reply:
This is written only aiming the second pos arg which is optional and is the error cause with the type `error?`. Correct me if I'm wrong.

Reply:
In error constructor expressions we can expect an optional second positional argument. Hence in that case, I have provided both `error?` typed completions and the expected named args in the case for the second argument. After the second argument, I only provide named args completions and as you suggested I have prioritised the snippets. As per the offline discussion, we agreed on the approach. Please unresolve and reply if there is any deviation.